src/main/java/seedu/address/commons/events/model/UserPersonChangedEvent.java

```diff
@@ -0,0 +1,24 @@
+package seedu.address.commons.events.model;
+
+import seedu.address.commons.events.BaseEvent;
+import seedu.address.model.person.UserPerson;
+
+//@@author bladerail
+/** Indicates the UserProfileManager in the model has changed*/
+public class UserPersonChangedEvent extends BaseEvent {
+
+    public final UserPerson userPerson;
+
+    public UserPersonChangedEvent(UserPerson data) {
+        this.userPerson = data;
```

Comment:
`this` is not necessary as there are no ambiguities between the constructor parameter `data` and the object attribute `userPerson`. While the use of `this` here is not strictly problematic, we avoid doing so for the sake of consistency in style.